addons/website_sale_referral/models/sale_order.py

```diff
@@ -0,0 +1,66 @@
+from odoo import models, fields, api, _, SUPERUSER_ID
+from ast import literal_eval
+
+
+class SaleOrder(models.Model):
+    _name = 'sale.order'
+    _inherit = ['sale.order', 'referral.mixin']
+
+    referred_email = fields.Char(string="Referral email", related='partner_id.email', description="The email used to identify the referred")
+    referred_name = fields.Char(string="Referral name", related='partner_id.name', description="The name of the referred")
+    referred_company = fields.Char(string="Referral company", compute='_compute_referred_company', store=False, description="The company of the referred")
```

Comment:
referred_company_name, as this is not a Many2one.

Reply:
Done